src/databricks/labs/remorph/reconcile/query_builder_refactored/sampling_query.py

```diff
@@ -0,0 +1,66 @@
+from pyspark.sql import DataFrame
+from sqlglot import select
+from sqlglot.expressions import Alias, Select, union
+
+from databricks.labs.remorph.reconcile.constants import SampleConfig
+from databricks.labs.remorph.reconcile.query_builder_refactored.base import QueryBuilder
+from databricks.labs.remorph.reconcile.query_builder_refactored.expression_generator import (
+    build_column,
+    build_literal,
+)
+
+
+def _union_concat(unions, result, cnt=0):
+    if len(unions) == 1:
+        return result
+    if cnt == len(unions) - 2:
+        return union(result, unions[cnt + 1])
+    cnt = cnt + 1
+    res = union(result, unions[cnt])
+    return _union_concat(unions, res, cnt)
+
+
+class SamplingQueryBuilder(QueryBuilder):
+
+    def build_query(self, df: DataFrame):
+        if self.layer == "source":
+            key_cols = sorted(self.join_columns)
+        else:
+            key_cols = sorted(self.table_conf.get_tgt_to_src_col_mapping(self.join_columns, self.layer))
+        keys_df = df.select(*key_cols)
+        with_clause = self._get_with_clause(keys_df)
+
+        cols = sorted((self.join_columns | self.select_columns) - self.threshold_columns - self.drop_columns)
+
+        cols_with_alias = [
+            build_column(this=col, alias=self.table_conf.get_tgt_to_src_col_mapping(col, self.layer)) for col in cols
+        ]
+
+        sql_with_transforms = self._add_transformations(cols_with_alias, self.source)
+        query_sql = select(*sql_with_transforms).from_(":tbl").where(self.filter)
+
+        return (
+            with_clause.with_(alias="src", as_=query_sql)
+            .select("src.*")
```

Comment:
don't use `*` operation use the actual column names.

Reply:
Updated.